lib/elixir/test/elixir/calendar/duration_test.exs

```diff
@@ -358,4 +358,75 @@ defmodule DurationTest do
     assert %Duration{microsecond: {-800_000, 0}} |> Duration.to_iso8601() == "PT0S"
     assert %Duration{microsecond: {-1_200_000, 2}} |> Duration.to_iso8601() == "PT-1.20S"
   end
+
+  test "to_string/1" do
+    assert Duration.to_string(%Duration{year: 1, month: 2, day: 3, hour: 4, minute: 5, second: 6}) ==
+             "1a 2mo 3d 4h 5min 6s"
+
+    assert Duration.to_string(%Duration{week: 3, hour: 5, minute: 3}) ==
+             "3wk 5h 3min"
+
+    assert Duration.to_string(%Duration{hour: 5, minute: 3}) ==
+             "5h 3min"
+
+    assert Duration.to_string(%Duration{year: 1, month: 2, day: 3}) ==
+             "1a 2mo 3d"
+
+    assert Duration.to_string(%Duration{hour: 4, minute: 5, second: 6}) ==
+             "4h 5min 6s"
+
```

Comment:
```suggestion
    assert Duration.to_string(%Duration{hour: 4, minute: 5, second: 6}) ==
             "4h 5min 6s"

    # When units with value 0 are surrounded by units > 0.
    assert Duration.to_string(%Duration{hour: 4, minute: 0, second: 6}) ==
             "4h 6s"

```

Reply:
We already test this since most examples do not consider weeks. :)